scripts/config.py

```diff
@@ -468,141 +504,124 @@ def _get_compatible_GCC(self, nvcc_version):
+                    return False
+        print(green('g++' + ' found in ' + prg))
+        self._set(Config.OPT_CXX_CUDA, prg)
+        return True
+
+    def _set_nvcc_lib_dir(self):
+        opt = Config.OPT_NVCC_LINKFLAGS
+        if not self.is_empty(opt):
+            return True
+
+        dirs = ['lib', 'lib64', 'targets/x86_64-linux/lib',
+                'lib/x86_64-linux-gnu']
+        # assume cuda_dir/bin/nvcc
+        locs = [os.path.dirname(os.path.dirname(self.get(Config.OPT_NVCC))),
+                os.environ.get('XMIPP_CUDA_LIB', ''),
+                os.environ.get('CUDA_LIB', ''),
+                os.environ.get('LD_LIBRARY_PATH', ''),
+                '/usr'
+                ]
+
+        def search():
+            for l in locs:
+                for d in dirs:
+                    tmp = os.path.join(l, d, 'libcudart.so')
+                    if os.path.isfile(tmp):
+                        return os.path.dirname(tmp)
+            return None
+
+        path = search()
+        if path is None:
+            print(yellow(
+                'WARNING: CUDA libraries (libcudart.so) not found. ' + self._get_help_msg()))
+            return False
+        # nvidia-ml is in stubs folder
+        self._set(opt, self._join_with_prefix(
+            [path, os.path.join(path, 'stubs')], '-L'))
+        return True
+
+    def _set_nvcc_flags(self, nvcc_version):
+        flags = ("--x cu -D_FORCE_INLINES -Xcompiler -fPIC "
+                 "-ccbin %(CXX_CUDA)s -std=c++11 --expt-extended-lambda "
+                 # generate PTX only, and SASS at the runtime (by setting code=virtual_arch)
+                 "-gencode=arch=compute_35,code=compute_35 "
+                 "-gencode=arch=compute_50,code=compute_50 "
+                 "-gencode=arch=compute_60,code=compute_60 "
+                 "-gencode=arch=compute_61,code=compute_61")
+        if nvcc_version >= 11:
+            flags = ("--x cu -D_FORCE_INLINES -Xcompiler -fPIC "
+                     "-ccbin %(CXX_CUDA)s -std=c++14 --expt-extended-lambda "
+                     # generate PTX only, and SASS at the runtime (by setting code=virtual_arch)
+                     "-gencode=arch=compute_60,code=compute_60 "
+                     "-gencode=arch=compute_61,code=compute_61 "
+                     "-gencode=arch=compute_75,code=compute_75 "
+                     "-gencode=arch=compute_86,code=compute_86")
+        self._set_if_empty(Config.OPT_NVCC_CXXFLAGS, flags)
+
+    def _set_CUDA(self):
+        def print_no_CUDA():
+            print(red("No valid compiler found. "
+                  "Skipping CUDA compilation.\n"))
+
+        if not self.is_empty(Config.OPT_CUDA):
+            print_no_CUDA()
+            return
+        if not self._set_nvcc():
+            print_no_CUDA()
+            return
+        nvcc_version, nvcc_full_version = self._get_CUDA_version(
+            self.get(Config.OPT_NVCC))
+        if nvcc_version == False:
```

Comment:
when can this happen?

Reply:
`_get_CUDA_version(nvcc)` returns `False, False` if does not find version

Reply:
yes, but can that happen and if yes, in which case?